test/index.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+require('chai').should();
+const expect = require('chai').expect;
+const proxyquire = require('proxyquire');
+const sinon = require('sinon');
+
+const EXAMPLE_SERVER_KEYS = {
+  public: 'BOg5KfYiBdDDRF12Ri17y3v+POPr8X0nVP2jDjowPVI/DMKU1aQ3OLdPH1iaakvR9/PHq6tNCzJH35v/JUz2crY=',
+  private: 'uDNsfsz91y2ywQeOHljVoiUg3j5RGrDVAswRqjP3v90='
+}
+
+const EXAMPLE_SALT = 'AAAAAAAAAAAAAAAAAAAAAA==';
+
+const EXAMPLE_INPUT = 'Hello, World.';
+const EXAMPLE_OUTPUT = 'CE2OS6BxfXsC2YbTdfkeWLlt4AKWbHZ3Fe53n5/4Yg==';
+
+const VALID_SUBSCRIPTION = {
+  endpoint: 'https://example-endpoint.com/example/1234',
+  keys: {
+    auth: '8eDyX_uCN0XRhSbY5hs7Hg==',
+    p256dh: 'BCIWgsnyXDv1VkhqL2P7YRBvdeuDnlwAPT2guNhdIoW3IP7GmHh1SMKPLxRf7x8vJy6ZFK3ol2ohgn_-0yP7QQA='
+  }
+}
+
+const INVALID_AUTH_SUBSCRIPTION = {
+  endpoint: 'https://example-endpoint.com/example/1234',
+  keys: {
+    auth: 'uCN0XRhSbY5hs7Hg==',
+    p256dh: 'BCIWgsnyXDv1VkhqL2P7YRBvdeuDnlwAPT2guNhdIoW3IP7GmHh1SMKPLxRf7x8vJy6ZFK3ol2ohgn_-0yP7QQA='
+  }
+}
+
+const INVALID_P256DH_SUBSCRIPTION = {
+  endpoint: 'https://example-endpoint.com/example/1234',
+  keys: {
+    auth: '8eDyX_uCN0XRhSbY5hs7Hg==',
+    p256dh: '6ZFK3ol2ohgn_-0yP7QQA='
+  }
+}
+
+const SUBSCRIPTION_NO_KEYS = {
+  endpoint: 'https://example-endpoint.com/example/1234'
+}
+
+const SALT_LENGTH = 16;
+const SERVER_PUBLIC_KEY_LENGTH = 65;
+
+describe('Test the Libraries Top Level API', function() {
+  describe('Test encrypt() method', function() {
+    it('should encrypt the message with a valid subscription', function() {
+      const library = require('../src/index.js');
+      const response = library.encrypt('Hello, World', VALID_SUBSCRIPTION);
+      Buffer.isBuffer(response.ciphertext).should.equal(true);
+      Buffer.isBuffer(response.salt).should.equal(true);
+      response.salt.should.have.length(SALT_LENGTH);
+      Buffer.isBuffer(response.serverPublicKey).should.equal(true);
+      response.serverPublicKey.should.have.length(SERVER_PUBLIC_KEY_LENGTH);
+    });
+
+    it('should throw an error due to subscription with no keys being passed in', function() {
+      const library = require('../src/index.js');
```

Comment:
Is there a particular reason to require the library in the tests rather than once at the top?